src/OpenSpaceToolkit/Astrodynamics/Trajectory/Orbit.cpp

```diff
@@ -1034,13 +1036,12 @@ Map<Index, Pass> Orbit::GeneratePassMap(const Array<State>& aStateArray, const I
 
     // [TBI] Deal w/ z_ECI always equal to 0.0 case (equatorial orbit)
 
-    Map<Index, Pass> passMap;
+    Array<Pair<Index, Pass>> passMap;
 
-    Integer revolutionNumber = anInitialRevolutionNumber;
-    Instant previousPassEndInstant = Instant::Undefined();
-
-    Index currentIndex = 0;
-    State const* previousStatePtr = nullptr;
+    if (aStateArray.getSize() < 2)
+    {
+        return passMap;
```

Comment:
```suggestion
        throw ostk::core::error::Runtime("Too few states to compute passes. n_states = {}", aStateArray.getSize());
```

Returning an uninitialized value seems kinda sketch. Maybe good to just error here instead?

Reply:
Especially since that's the way we handle states not being in chrono order on line 1050 below

Reply:
Yeah I went back and forth on this. I think raising an error is fair, users can handle it.